lib/srthelper.rb

```diff
@@ -0,0 +1,141 @@
+		cf =File.read(input_file)
+		data = JSON.parse(cf)
+		sentence_list = []
+		str = ""
+
+		prev_end_time = nil
+		sentence_start_time = nil
+		sentence_end_time = nil
+		data["results"]["items"].each do |item|
+			content = item["alternatives"][0]["content"]
+			#puts content
+
+			if item["type"].eql?("pronunciation")
+				start_time = item["start_time"]
+				end_time = item["end_time"]
+				if prev_end_time.nil?
+					prev_end_time = end_time
+					sentence_start_time = start_time
+					sentence_end_time = end_time
+					str = content
+				else
+					if prev_end_time == start_time
+						str = "#{str} #{content}"
+						sentence_end_time = end_time
+						prev_end_time = end_time
+					else
+						# End of Sentence
+						# So this word should come in the next sentence
+						sentence_object = {"sentence" => str , "start_time" => sentence_start_time , "end_time" => sentence_end_time, "punctuation" => false}
+						sentence_list << sentence_object
+						str = content
+						sentence_start_time = start_time
+						sentence_end_time = end_time
+						prev_end_time = end_time
+					end
+				end
+			else
+				str = "#{str}#{content}"
+				sentence_object = {"sentence" => str , "start_time" => sentence_start_time , "end_time" => sentence_end_time, "punctuation" => true}
+				sentence_list << sentence_object
+				str = ""
+				prev_end_time = nil
+			end
+		end
+
+		# Merge all the sentence based on 10 milli second gap and Punctuation
+	
+		new_sentence_list = []
+		sentence = ""
+		sentence_start_time = nil
+		sentence_end_time = nil
+		sub_list = []
+		sentence_list.each do |sent|
+			content = sent["sentence"]
+			if sentence_start_time.nil?
+				sentence = content
+				sentence_start_time = sent["start_time"]
+				sentence_end_time = sent["end_time"]
+				sub_list << sent
+			else
+				sentence_start_time_f = sentence_start_time.to_f
+				sentence_end_time_f = sentence_end_time.to_f
+				start_time_f = sent["start_time"].to_f
+				end_time_f = sent["end_time"].to_f
+				cont_diff = start_time_f - sentence_end_time_f
+				total_sec_len = end_time_f - sentence_start_time_f
+				
+				if cont_diff > SENTENCE_CONTINUATION_DIFF || (total_sec_len > SINGLE_SRT_SECONDS_LENGTH && word_count(content) < CONTINUATION_SENTENCE_OVERLOAD)
+					# Need to end the sentence
+					sentence_object = {"sentence" => sentence , "start_time" => sentence_start_time , "end_time" => sentence_end_time, "sub_list" => sub_list}
+					new_sentence_list << sentence_object
+					sentence = content
+					sentence_start_time = sent["start_time"]
+					sentence_end_time = sent["end_time"]
+					sub_list = []
+				else
+					sentence = "#{sentence} #{content}"
+					sentence_end_time = sent["end_time"]
+					sub_list << sent
+				end
+
+			end
+		end
+
+		srt_generator(new_sentence_list, output_file)
+	end
+
+
+	def word_count(word)
+		str = word.strip
+		str_arr = str.split(" ")
+		return str_arr.size
+	end
+
+	def convert_time_srt_fmt(val)
```

Comment:
Can you apply our usual Code formatter ?

Reply:
Done